test/cbmc/proofs/DNS/DNSgetHostByName_a/DNSgetHostByName_a_harness.c

```diff
@@ -90,24 +162,48 @@ void func( const char * pcHostName,
 {
 }
 
+
+/** A list of all network end-points.  Each element has a next pointer. */
+extern struct xNetworkEndPoint * pxNetworkEndPoints;
+
+/** A list of all network interfaces: */
+extern struct xNetworkInterface * pxNetworkInterfaces;
+
+
 /****************************************************************
 * The proof for FreeRTOS_gethostbyname_a.
 ****************************************************************/
-
 void harness()
 {
     size_t len;
+    __CPROVER_assume( ( len <= MAX_HOSTNAME_LEN ) && ( len > 0 ) );
 
-    __CPROVER_assume( len <= MAX_HOSTNAME_LEN );
-    char * pcHostName = safeMalloc( len );
+    /* Arbitrarily assign a proper memory or NULL to the hostname. */
+    char * pcHostName = nondet_bool() ? safeMalloc( len ) : NULL;
```

Comment:
I'm pretty sure safeMalloc is already doing this nondeterministic choice of NULL.

Reply:
Nice catch!
I will change that to malloc then just to be inline with the code.